pkg/route2httpproxy/route2httpproxy_test.go

```diff
@@ -0,0 +1,111 @@
+package route2httpproxy
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"path/filepath"
+	"testing"
+
+	route "github.com/openshift/api/route/v1"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	core "k8s.io/api/core/v1"
+)
+
+func TestDeployment(t *testing.T) {
+	type output struct {
+		kind       string
+		apiVersion string
+		fqdn       string
+	}
+
+	tests := []struct {
+		name         string
+		routeInput   string
+		serviceInput string
+		want         output
+		domain       string
+	}{
+		{
+			"Test config map with Rolling Strategy",
+			"route_with_tls.json",
+			"service-input.json",
+			output{
+				kind:       "HTTPProxy",
+				apiVersion: "projectcontour.io/v1",
+				fqdn:       "my-nginx-nginx-example.migrator.servicemesh.biz",
+			},
+			"*.migrator.servicemesh.biz",
+		},
+		{
+			"Test config map with Rolling Strategy",
+			"route_without_tls.json",
+			"service-input.json",
+			output{
+				kind:       "HTTPProxy",
+				apiVersion: "projectcontour.io/v1",
+				fqdn:       "my-nginx-nginx-example.migrator.servicemesh.biz",
+			},
+			"*.migrator.servicemesh.biz",
+		},
+	}
+
+	for _, tc := range tests {
+		routeInput, serviceInput := newMutatorFromFileData(t, tc.routeInput, tc.serviceInput, tc.name)
+		hp, service, _ := Mutate(tc.name, logrus.New(), routeInput, serviceInput, tc.domain)
```

Comment:
i think `service` should be `secret` here since that's the second return value of this function and then down on lines 76/77 the TLS keys/certs are being compared